main.go

```diff
@@ -462,3 +487,207 @@ func stopStalePipelines (client *gitlab.Client, vars []*gitlab.PipelineVariable)
+// starting at the branch in which to sync, with the name 'PRBranchName'
+// and merges this with the OS equivalent of 'branchName'.
+func createPRBranchOnEnterprise(repo, branchName, PRBranchName string) error {
+
+	tmpdir, err := ioutil.TempDir("", repo)
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpdir)
+
+	gitcmd := exec.Command("git", "init", ".")
+	gitcmd.Dir = tmpdir
+	out, err := gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	gitcmd = exec.Command("git", "remote", "add", "opensource", "git@github.com:mendersoftware/"+repo+".git")
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	gitcmd = exec.Command("git", "remote", "add", "enterprise", "git@github.com:mendersoftware/"+repo+"-enterprise"+".git")
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// Fetch the branch which we are going to sync
+	gitcmd = exec.Command("git", "fetch", "opensource", branchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// Fetch the Enterprise branch in which to merge into, and create the PR branch
+	gitcmd = exec.Command("git", "fetch", "enterprise", branchName+":"+PRBranchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// Checkout the enterprise PR branch
+	gitcmd = exec.Command("git", "checkout", PRBranchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// Merge the branches
+	mergeMessage := fmt.Sprintf("Merge branch '%s' into '%s'", "opensource/"+branchName, "enterprise/"+PRBranchName)
+	gitcmd = exec.Command("git", "merge", "-m", mergeMessage, "opensource/"+branchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	log.Debugf("git merge output: %s", out)
+
+	switch string(out) {
+	case "Already up to date.\n":
+		return errors.New("The branches seem to already be in sync, which they are not supposed to be, after a merge to OS!")
+	}
+
+	// Push the branch to enterprise
+	gitcmd = exec.Command("git", "push", "--set-upstream", "enterprise", PRBranchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	log.Infof("Merged branch: opensource/%s/%s into enterprise/%[1]s/%s in the Enterprise repo", repo, branchName, PRBranchName)
```

Comment:
I didn't know about this syntax for formatting (the `%[1]s`). Nice

Reply:
It reuses the first argument to format.